### PR TITLE
feat: notification connector

### PR DESCRIPTION
**Pull request by bwpikaard**

### CODE REVIEW

File: common/src/service-connectors/notification/schemas/SendNotification.schema.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import {z} from "zod";
+
+import {
+    SendDirectMessage_Request,
+    SendGuildTextMessage_Request,
+    SendWebhookMessage_Request,
+} from "../../bot";
+
+export enum NotificationType {
+    RANKDOWN = "RANKDOWN",
```
Comment (zachstence) [inline common/src/service-connectors/notification/schemas/SendNotification.schema.ts:10]: Don't we also have notifications for scrims?

File: common/src/service-connectors/notification/schemas/SendNotification.schema.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    TEST = "TEST",
```
Comment (zachstence) [inline common/src/service-connectors/notification/schemas/SendNotification.schema.ts:11]: Remove `TEST` stuff

File: microservices/notification-service/src/notification/notification.service.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        private readonly botService: BotService,
+    ) {}
+
+    async sendNotification(data: NotificationInput<NotificationEndpoint.SendNotification>): Promise<NotificationOutput<NotificationEndpoint.SendNotification>> {
+        if (data.payload) {
+            const notificationPayload = {
+                id: randomUUID(),
+                type: data.type,
+                expiration: data.expiration,
+                ...data.payload,
```
Comment (zachstence) [inline microservices/notification-service/src/notification/notification.service.ts:29]:
```suggestion
                ...data.payload,
```

File: common/src/service-connectors/notification/schemas/SendNotification.schema.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    expiration: z.date().nullable()
+        .optional(),
+    payload: z.object({}).nullable()
+        .optional(),
+    notification: z.union([
+        NotificationGuildTextMessageSchema,
+        NotificationDirectMessageSchema,
+        NotificationWebhookMessageSchema,
+    ]).nullable()
+        .optional(),
```
Comment (zachstence) [inline common/src/service-connectors/notification/schemas/SendNotification.schema.ts:47]: Why would this be nullable or optional? What's the point of sending a notification without a notification?

File: common/src/service-connectors/notification/schemas/SendNotification.schema.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    payload: z.object({
+        playerId: z.number(),
+        skillGroupId: z.number(),
+        salary: z.number(),
+    }),
+    notification: NotificationDirectMessageSchema,
+});
+
+export type RankdownNotification = z.infer<typeof RankdownNotificationSchema>;
+export type RankdownNotificationPayload = RankdownNotification["payload"];
```
Comment (zachstence) [inline common/src/service-connectors/notification/schemas/SendNotification.schema.ts:62]: Put this in a separate file rankdown-notification.schema.ts so its easier to find when searching the repo

**Review by zachstence — CHANGES_REQUESTED** [review]

File: common/src/service-connectors/notification/schemas/SendNotification.schema.ts (hunk 1 above)
Comment (bwpikaard) [reply common/src/service-connectors/notification/schemas/SendNotification.schema.ts:10]: Those are handled by events.

File: microservices/notification-service/src/notification/notification.service.ts (hunk 3 above)
Comment (bwpikaard) [reply microservices/notification-service/src/notification/notification.service.ts:29]: `data.payload` doesn't contain type and expiration

File: common/src/service-connectors/notification/schemas/SendNotification.schema.ts (hunk 4 above)
Comment (bwpikaard) [reply common/src/service-connectors/notification/schemas/SendNotification.schema.ts:47]: Good point.

File: microservices/notification-service/src/notification/notification.service.ts (hunk 3 above)
Comment (zachstence) [reply microservices/notification-service/src/notification/notification.service.ts]: Am brick

**Review by zachstence — APPROVED** [review]

**Pull request merged**